internal/backends/ent/schema/annotation.go

```diff
@@ -25,18 +26,33 @@ func (Annotation) Mixin() []ent.Mixin {
 
 func (Annotation) Fields() []ent.Field {
 	return []ent.Field{
+		field.String("node_id").Optional(),
 		field.String("name"),
 		field.String("value"),
 		field.Bool("is_unique").
 			Default(false),
 	}
 }
 
+func (Annotation) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("node", Node.Type).
+			Ref("annotations").
+			Unique().
+			Field("node_id"),
+	}
+}
+
 func (Annotation) Indexes() []ent.Index {
 	return []ent.Index{
+		index.Fields("node_id", "name", "value").
+			Unique().
+			Annotations(entsql.IndexWhere("node_id != ''")).
```

Comment:
Could `node_id` potentially be `null`? Is there a way in SQLite syntax to check for both empty string and `null` (falsy) simultaneously?